tests/flow_function_parentheses/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -235,6 +280,21 @@ interface F {
 
 type ExtractType = <A>(B<C>) => D;
 
+type T = ?() => A;
+
+type T = ?(() => A) | B;
+
+type T = ?() => A | B;
```

Comment:
Would it make sense to explicitly disambiguate this to `?() => (A | B)`?

Reply:
It’s not clear to me, but I don’t use Flow so I’ll leave the decision up to those who do.

Reply:
If we end up deciding to wrap it in parentheses, I'd prefer to wrap the whole function. e.g.
```js
type T = ?(() => A | B)
```

This is consistent with how JS functions work:
```js
+(() => a || b);

+() => a || b; // SyntaxError
```

Reply:
My personal anecdote: I can't remember the operator precedence for `|`, so I actually thought that type was `(?() => A) | (B)`. So parenthesis would be helpful for me.

Reply:
I have a local implementation which puts the parentheses around `A | B` in `() => A | B`, but some snapsot tests look a bit strange to me.

e.g.
```diff
  then<U>(
-    onFulfill?: ?(value: R) => Promise<U> | ?U,
-    onReject?: ?(error: any) => Promise<U> | ?U
+    onFulfill?: ?(value: R) => (Promise<U> | ?U),
+    onReject?: ?(error: any) => (Promise<U> | ?U)
  ): Promise<U>;
```

Can we merge this PR as-is (since it fixes an actual bug with prettier, which changes the meaning of the code) and discuss about `() => A | B` in a new issue?

Reply:
> discuss about `() => A | B` in a new issue?

Agreed